relay-server/src/actors/project.rs

```diff
@@ -867,9 +880,18 @@ impl Project {
 
         match self.expiry_state() {
             // If the new state is invalid but the old one still usable, keep the old one.
-            ExpiryState::Updated(old) | ExpiryState::Stale(old) if state.invalid() => state = old,
+            //This also sets the next fetch attempt time.
+            ExpiryState::Updated(old) | ExpiryState::Stale(old) if state.invalid() => {
+                state = old;
+                self.next_fetch_attempt = Instant::now().checked_add(self.backoff.next_backoff());
+            }
             // If the new state is valid or the old one is expired, always use the new one.
-            _ => self.state = Some(state.clone()),
+            // And also reset the backoff at this point.
+            _ => {
```

Comment:
What happens when `self.expiry_state()` is `Expired`, but the new `state` is `invalid()`? Shouldn't we then also set a `next_fetch_attempt`?

Reply:
Currently (and the behaviour we had before) once the old state expires we save anything we get from upstream - this case the upstream is alive, just delivers the invalid state.  Since we cannot use the old state either, we will just try to request the new one soon-ish again. 
Do you think we have to re-think this behaviour in this PR?